ml_engine/online_prediction/predict.py

```diff
@@ -0,0 +1,184 @@
+            expected tensors
+        version: str, version of the model to target.
+    Returns:
+        Mapping[str: any]: dictionary of prediction results defined by the
+            model.
+    """
+    # Create the ML Engine service object.
+    # To authenticate set the environment variable
+    # GOOGLE_APPLICATION_CREDENTIALS=<path_to_service_account_file>
+    service = googleapiclient.discovery.build('ml', 'v1beta1')
+    name = 'projects/{}/models/{}'.format(project, model)
+
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': instances}
+    ).execute()
+
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_json]
+
+
+# [START predict_tf_records]
+def predict_tf_records(project,
+                       model,
+                       example_bytes_list,
+                       version=None):
+    """Send protocol buffer data to a deployed model for prediction.
+
+    Args:
+        project (str): project where the Cloud ML Engine Model is deployed.
+        model (str): model name.
+        example_bytes_list ([str]): A list of bytestrings representing
+            serialized tf.train.Example protocol buffers. The contents of this
+            protocol buffer will change depending on the signature of your
+            deployed model.
+        version: str, version of the model to target.
+    Returns:
+        Mapping[str: any]: dictionary of prediction results defined by the
+            model.
+    """
+    service = googleapiclient.discovery.build('ml', 'v1beta1')
+    name = 'projects/{}/models/{}'.format(project, model)
+
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': [
+            {'b64': base64.b64encode(example_bytes)}
+            for example_bytes in example_bytes_list
+        ]}
+    ).execute()
+
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_tf_records]
+
+
+# [START census_to_example_bytes]
+def census_to_example_bytes(json_instance):
+    """Serialize a JSON example to the bytes of a tf.train.Example.
+    This method is specific to the signature of the Census example.
+    See: https://cloud.google.com/ml-engine/docs/concepts/prediction-overview
+    for details.
+
+    Args:
+        json_instance (Mapping[str: any]): representing data to be serialized.
+    Returns:
+        str: A string as a container for the serialized bytes of
+            tf.train.Example protocol buffer.
+    """
+    import tensorflow as tf
+    feature_dict = {}
+    for key, data in json_instance.iteritems():
+        if isinstance(data, str) or isinstance(data, unicode):
```

Comment:
`isinstance(data, (str, unicode))`, but this will fail on Python 3 because `unicode` doesn't exist. You should consider using `six.string_type` or making this the else clause.